pkg/lib/packagekit.js

```diff
@@ -479,3 +521,212 @@ export function getBackendName() {
     return call("/org/freedesktop/PackageKit", "org.freedesktop.DBus.Properties",
                 "Get", ["org.freedesktop.PackageKit", "BackendName"]);
 }
+
+/**
+ * Refresh PackageKit Cache
+ * @param {boolean} force - force refresh the cache (expensive)
+ * @param {*} progress_cb - progress callback
+ */
+export function refresh(force = false, progress_cb) {
+    return cancellableTransaction("RefreshCache", [force], progress_cb);
+}
+
+/**
+ * Remove packages by their names.
+ *
+ * @param {?string[] | undefined} pkgnames - packages to remove
+ * @param {?() => void} progress_cb - progress callback
+ */
+export async function remove_packages(pkgnames, progress_cb) {
+    const ids = [];
+
+    const flags = Enum.FILTER_NOT_SOURCE | Enum.FILTER_INSTALLED | Enum.FILTER_NOT_SOURCE;
+    await cancellableTransaction("Resolve", [flags, pkgnames], progress_cb,
+                                 {
+                                     Package: (_info, package_id) => ids.push(package_id),
+                                 });
+
+    if (ids.length === 0)
+        return Promise.resolve();
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22498-3725f1ba-20251117-084332-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)